File: python/code_examples/02-collections/main.py
# list / tuple / dict / set
nums = [1, 2, 3]
point = (10, 20)  # immutable tuple
user = {"id": 1, "name": "Ada"}
unique = {1, 2, 3, 3}  # {1, 2, 3}

# dict access
user["role"] = "dev"
role = user.get("role", "guest")


def main() -> None:
    print(nums, point, user, unique, role)


if __name__ == "__main__":
    main()

File: python/code_examples/02-collections/main.js
// Array, object-as-map, Set
const nums = [1, 2, 3];
const point = Object.freeze([10, 20]); // not a real tuple — arrays are mutable
const user = { id: 1, name: "Ada" };
const unique = new Set([1, 2, 3, 3]);

user.role = "dev";
const role = user.role ?? "guest";

function main() {
  console.log(nums, point, user, [...unique], role);
}

main();
